### ISSUE TITLE
(Docs) OpenCV installation on linux - libhdf5 dependency

### ISSUE THREAD

**Issue by Munawwar** [opencv/opencv#7230]
Hi. libhdf5-dev is necessary to build opencv imgproc module successfully on linux (at least on Ubuntu 16.04). Please add the dependency to the docs here -> http://docs.opencv.org/3.1.0/d7/d9f/tutorial_linux_install.html.

**Comment by mshabunin** [comment]
@Munawwar , I was able to build latest OpenCV master version on Ubuntu 16.04 without _libhdf5-dev_ package installed. Please, describe your problem more precisely.

**Comment by Munawwar** [comment]
Ah, I can't replicate the issue consistently.

I should have noted down the build errors. I remember seeing an error message on one of the imgproc modules. I now removed libhdf5-dev, did a `make clean` and hit the following error on a contrib module:
```
opencv_contrib/modules/hdf/src/hdf5.cpp:37:18: fatal error: hdf5.h: No such file or directory
compilation terminated.)
```
I removed the build/ directory completely and tried rebuilding and the build succeeds :confused:. Heisenbug?

Let's close the issue for now, and reopen it if someone else can replicate the issue.
